### ISSUE TITLE
[figma][help] Create a Figma script that creates component sets for icons

### ISSUE THREAD

**Issue by adrianmanea** [mui/mui-design-kits#261]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Hey there! I'm reaching out because I could really use some advice, especially if anyone here has experience with the Figma API.

I've been picturing a unified library, something akin to [Google's community library](https://www.figma.com/community/file/1014241558898418245/material-design-icons). It would be fantastic to have all styles as component variants readily available.

So, I'm thinking we could use a script. This script would take the selected frames and create component sets from the ones that match. Any thoughts or guidance would be greatly appreciated!

### Examples 🌈

Given the following Frames `addchart_24px.svg`, `addchart_sharp_24px.svg` [...] outputs a component set with the variants: `filled` and `sharp` respectively. Note that if no matching style found it defaults to `filled`.

Basically automates the following process:

https://github.com/mui/mui-design-kits/assets/5917515/5e03c98e-ed85-4e3c-9aac-58ae8f7d5bd8
